src/SmartComponents/RunTaskModal/ParameterDropdown.js

```diff
@@ -0,0 +1,55 @@
+import {
+  FormGroup,
+  FormHelperText,
+  FormSelect,
+  FormSelectOption,
+  HelperText,
+  HelperTextItem,
+} from '@patternfly/react-core';
+import React, { useState } from 'react';
+import propTypes from 'prop-types';
+
+const ParameterDropdown = ({
+  parameter,
+  updateParameters,
+  customDescription = undefined,
+}) => {
+  const [parameterValue, setParameterValue] = useState(parameter.default);
```

Comment:
Do we really need to keep this internal component state? Isn't `parameter` object already containing the current value for the parameter? You can fully delegate the state management to the parent component through this prop I believe.

Reply:
The parameter object only contains the current value if its different from the default value.  So the parameter doesn't have a value attribute until its changed from the default value.  So I the component needs its own state to keep track of whether the current value for the component is either the default value or the changed value.  I hope that makes sense.

Perhaps we could change each parameter so it automatically includes a value attribute, then we wouldn't need to maintain its state internally.